deepinterpolation/generator_collection.py

```diff
@@ -1038,6 +1039,7 @@ def __init__(self, json_path):
             self.frame_data_location = json.load(json_handle)
 
         self.lims_id = list(self.frame_data_location.keys())
+        self.lims_id.sort()
```

Comment:
Why is this line needed?

Reply:
When I was testing my changes, I wanted to make sure that every time I ran the code, I got the frames in the same order (also: I just don't like it when I can't guarantee the order in which things will get processed ;)

It shouldn't affect the performance of the trainer, since the generator class loops over videos, then frames i.e. it supplies
```
movie0, frame0
movie1, frame0
movie2, frame0
....
movie0, frame1
movie1, frame1
movie2, frame1
....
```

Reply:
This sorts the experiment ids, not the frames, correct? I want to make sure that each batch is randomly sampled, since experiment ids that are close together tend to be similar, and each batch should be randomly drawn. It's hard to tell where/if shuffling occurs somewhere downstream.

Reply:
Are we sure that without the sort the order of the keys will be what is specified in the json file? If json loads a `dict`, rather than an `OrderedDict`, I don't think we can be guaranteed anything? My understanding of `dict.keys()` is that it is not necessarily returned in any specified order.

Reply:
Sorry. Apparently they fixed that in python 3.7 so that dicts to preserve insertion order

https://stackoverflow.com/questions/1867861/how-to-keep-keys-values-in-same-order-as-declared

Reply:
FYI an alternative would be to read in the list, sort it, then randomize it with a random number generator that the user specified at run time. Then the onus wouldn't be on the user to remember to randomize file orders when creating the input JSON (I often scan through files in alphabetical order just to maintain my own sanity).

Reply:
Agreed, onus should not be on the user to randomize the inputs.

Reply:
You are welcome to randomize it in the generator. But I am still worried about sorting. :-)

Reply:
@jeromelecoq
This is what I am describing

https://github.com/AllenInstitute/deepinterpolation/pull/83/commits/9161114c7cfcb9a966cbdaf39594731781f0c9f2

The lims_ids are sorted upon ingest, but immediately randomized using a random number generator with a user-specified seed. Allowing the user to specify the seed gives them the ability to reproduce results across runs (assuming, of course, that tensorflows output depends exactly on the order in which the data is provided to it; I cannot speak to that)

What do you think?